baker/siteRenderers.tsx

```diff
@@ -215,6 +217,60 @@ export const renderPost = async (
 export const renderFrontPage = async () => {
     const entries = await getEntriesByCategory()
     const posts = await getBlogIndex()
+
+    const NUM_FEATURED_POSTS = 6
+
+    /**
+     * A frontPageConfig should specify a list of
+     * articles to feature in the featured work block,
+     * and which position they should be placed in.
+     *
+     * Note that this can be underspecified, so some positions
+     * may not be filled in.
+     *
+     */
+
+    let featuredWork: IndexPost[]
+    try {
+        const frontPageConfigGdoc = new Gdoc(GDOCS_HOMEPAGE_CONFIG_DOCUMENT_ID)
+        await frontPageConfigGdoc.getEnrichedArticle()
+        const frontPageConfig: any = frontPageConfigGdoc.content
+        const featuredPosts: { slug: string; position: number }[] =
+            frontPageConfig.featuredPosts
+
+        // Generate the candidate posts to fill in any missing slots
+        const slugs = featuredPosts.map((d) => d.slug)
+        const filteredPosts = posts.filter((post) => {
+            return !slugs.includes(post.slug)
+        })
+
+        /**
+         * Create the final list of featured work by merging the
+         * manually curated list of posts and filling in any empty
+         * positions with the latest available posts, while avoiding
+         * adding any duplicates.
+         */
+        let missingPosts = 0
+        featuredWork = [...new Array(NUM_FEATURED_POSTS)]
+            .map((_, i) => i)
+            .map((idx) => {
+                const manuallySetPost = featuredPosts.find(
+                    (d) => +d.position === idx + 1
```

Comment:
I wonder if we have any other index data that the authors write. Hopefully it's index-starts-at-1 everywhere 😬